source/write/transactions.txt

```diff
@@ -0,0 +1,164 @@
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+The documents in this collection are modeled by the following {+language+} data class:
+
+.. literalinclude:: /includes/write/transaction.kt
+   :start-after: start-data-class
+   :end-before: end-data-class
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+Methods
+-------
+
+Create a ``ClientSession`` by using the ``startSession()`` method on your ``MongoClient``
+instance. You can then modify the session state by using the methods provided by
+the ``ClientSession``. The following table describes the methods you can use to
+manage your transaction:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``startTransaction()``
+     - | Starts a new transaction, configured with the given options, on
+         this session. Returns an error if there is already
+         a transaction in progress for the session. To learn more about
+         this method, see the :manual:`startTransaction() page
+         </reference/method/Session.startTransaction/>`  in the Server manual.
+       |
+       | **Parameter**: ``TransactionOptions``
+
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. Returns an
+         error if there is no active transaction for the session or the
+         transaction has been committed or ended. To learn more about
+         this method, see the :manual:`abortTransaction() page
+         </reference/method/Session.abortTransaction/>`  in the Server manual.
+       |
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or if the
+         transaction was ended. To learn more about
+         this method, see the :manual:`commitTransaction() page
+         </reference/method/Session.commitTransaction/>`  in the Server manual.
+
+   * - ``withTransaction()``
+     - | Starts a transaction on this session and runs the given function within 
+         a transaction.
+       |
+       | **Parameters**: transaction body function, ``TransactionOptions``
+
+Example
+-------
+
+The following example demonstrates how to create a session, create a
+transaction, and insert documents into a collection in one transaction
+through the following steps:
+
+1. Create a session from the client by using the ``startSession()`` method.
+#. Use the ``withTransaction()`` method to start a transaction.
+#. Insert multiple documents into the ``restaurants`` collection. The 
+   ``withTransaction()`` method runs the insert operation and commits the 
+   transaction. If any operation results in errors, ``withTransaction()`` 
+   cancels the transaction.
```

Comment:
```suggestion
#. Define the ``xxx()`` method to insert multiple documents into the ``restaurants`` collection. The 
   ``withTransaction()`` method runs the insert operation and commits the 
   transaction. If any operation results in errors, ``withTransaction()`` 
   cancels the transaction.
```

Reply:
Updated and reordered the steps to better match the code snippet.